pkg/pub_package_reader/lib/pub_package_reader.dart

```diff
@@ -119,7 +126,10 @@ Future<PackageSummary> summarizePackageArchive(
     return PackageSummary(issues: issues);
   }
 
-  final pubspecContent = await tar.readContentAsString(pubspecPath);
+  final pubspecContent = overridePubspecContentIfNeeded(
+    content: await tar.readContentAsString(pubspecPath),
+    created: created ?? DateTime.now().toUtc(),
```

Comment:
```suggestion
    created: created ?? clock.now().toUtc(),
```

If we want it to be optional..